mmda/predictors/hf_predictors/bibentry_predictor.py

```diff
@@ -0,0 +1,217 @@
+from enum import Enum
+import re
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel
+from transformers import AutoConfig, AutoTokenizer, AutoModelForTokenClassification
+from unidecode import unidecode
+
+
+class Label(Enum):
+    MISC = 0
+
+    CITATION_NUMBER = 1
+
+    AUTHOR_START = 2
+    AUTHOR_MIDDLE = 3
+    AUTHOR_END = 4
+
+    ISSUED_DAY = 5
+    ISSUED_MONTH = 6
+    ISSUED_YEAR = 7
+
+    TITLE = 8
+    JOURNAL = 9
+    PUBLISHER = 10
+    VOLUME = 11
+    ISSUE = 12
+    PAGE = 13
+    URL = 14
+    DOI = 15
+    EVENT = 16
+    ISBN = 17
+
+
+class StringWithSpan(BaseModel):
+    content: str
+    start: int  # inclusive
+    end: int  # exclusive
+
+
+class PredictionWithSpan(BaseModel):
+    citation_number: Optional[StringWithSpan]
+    authors: Optional[List[StringWithSpan]]
+    title: Optional[StringWithSpan]
+    journal_venue_or_event: Optional[StringWithSpan]
+    year: Optional[StringWithSpan]
+    doi: Optional[StringWithSpan]
+    url: Optional[StringWithSpan]
+
+
+class BibEntryPredictor:
```

Comment:
I've been calling it BibEntryParser, but since it's under `hf_predictor` I renamed it to `BibEntryPredictor`. Which name do you prefer `BibEntryParser` or `BibEntryPredictor`?